packages/js/src/components/DefaultSeoDataAlert.js

```diff
@@ -0,0 +1,135 @@
+import { Fragment, useCallback, useMemo } from "@wordpress/element";
+import { applyFilters } from "@wordpress/hooks";
+import { Slot } from "@wordpress/components";
+import { Button } from "@yoast/components";
+import { safeCreateInterpolateElement } from "../helpers/i18n";
+
+/**
+ * Renders the Default SEO Data Alert.
+ *
+ * @param {Object} isSeoDataDefault Whether SEO titles and meta descriptions are default.
+ *
+ * @returns {wp.Element} The Default SEO Data Alert.
+ */
+export default function DefaultSeoDataAlert( {
+	isSeoDataDefault,
+} ) {
+	const postType = select( "yoast-seo/editor" ).getPostType();
+	const isTitlesDefault = useMemo( () => {
+		return isSeoDataDefault?.isAllTitlesDefault || false;
+	}, [ isSeoDataDefault ] );
+
+	const isDescriptionsDefault = useMemo( () => {
+		return isSeoDataDefault?.isAllDescriptionsDefault || false;
+	}, [ isSeoDataDefault ] );
+
+	const showAlert = useMemo( () => {
+		if ( postType !== "post" ) {
+			return false;
+		}
+
+		return isTitlesDefault || isDescriptionsDefault;
+	}, [ postType, isTitlesDefault, isDescriptionsDefault ] );
+
+
+	const seoDataNames = useMemo( () => {
+		if ( isTitlesDefault && isDescriptionsDefault ) {
+			return __( "SEO titles and meta descriptions", "wordpress-seo" );
+		} else if ( isTitlesDefault ) {
+			return __( "SEO titles", "wordpress-seo" );
+		} else if ( isDescriptionsDefault ) {
+			return __( "meta descriptions", "wordpress-seo" );
+		}
+	}, [ isTitlesDefault, isDescriptionsDefault ] );
+
+	const message = useMemo( () => sprintf(
+		/* translators: %1$s expand to "SEO title" or "meta description" or both. */
+		__( "Stand out in the search results and attract more visitors by adding custom %1$s.", "wordpress-seo" ),
+		seoDataNames
+	), [ seoDataNames ] );
+
+	const proTip = useMemo( () => safeCreateInterpolateElement(
+		sprintf(
+			/* translators: %1$s, %2$s expand to strong tags. %3$s, %4$s expand to emphasis tags. %5$s expands to "SEO title" or "meta description" */
+			__(
+				"%1$sPro tip%2$s: Use %3$sAI Generate%4$s for quick optimized %5$s.",
+				"wordpress-seo"
+			),
+			"<strong>",
+			"</strong>",
+			"<em>",
+			"</em>",
+			seoDataNames
+		),
+		{
+			strong: <strong />,
+			em: <em />,
+		}
+	), [ seoDataNames ] );
+
+	const titleButtons = useMemo( () => {
+		if ( ! isTitlesDefault ) {
+			return [];
+		}
+
+		if ( isDescriptionsDefault ) {
+			return [];
+		}
+
+		return applyFilters( "yoast.replacementVariableEditor.additionalButtons", [], { fieldId: "yoast-google-preview-pre-publish", type: "title" } );
+	}, [ isTitlesDefault, isDescriptionsDefault ] );
+
+	const descButtons = useMemo( () => {
+		if ( ! isDescriptionsDefault ) {
+			return [];
+		}
+
+		return applyFilters( "yoast.replacementVariableEditor.additionalButtons", [], { fieldId: "yoast-google-preview-pre-publish", type: "description" } );
+	}, [ isTitlesDefault, isDescriptionsDefault ] );
+
+	const { openGeneralSidebar } = useDispatch(
+		"core/edit-post"
+	) || {};
```

Comment:
Maybe it's nicer to write it:
```js
const openGeneralSidebar = useDispatch("core/edit-post")?.openGeneralSidebar;
```

Reply:
Added [here](https://github.com/Yoast/wordpress-seo/pull/22623/commits/c4094e230e418f3d6f7f2e3af782255ddbe019aa).